pkg/ruler/metrics.go

```diff
@@ -0,0 +1,36 @@
+package ruler
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var samplesEvicted = promauto.NewCounterVec(prometheus.CounterOpts{
+	Namespace: "loki",
+	Name:      "recording_rules_samples_evicted_total",
+	Help:      "Number of samples evicted from queue; queue is full!",
+}, []string{"user_id", "group_key"})
+
+var samplesQueuedTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+	Namespace: "loki",
+	Name:      "recording_rules_samples_queued_total",
+	Help:      "Number of samples queued in total.",
+}, []string{"user_id", "group_key"})
+
+var samplesQueued = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	Namespace: "loki",
+	Name:      "recording_rules_samples_queued_current",
+	Help:      "Number of samples queued to be remote-written.",
+}, []string{"user_id", "group_key"})
+
+var samplesQueueCapacity = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
This is a per tenant limit, but the metric is partitioned by group. Can we make this only partitioned by tenant?

Reply:
Great point, will do.